fe/src/main/java/org/apache/doris/alter/MaterializedViewHandler.java

```diff
@@ -716,7 +716,6 @@ public void processBatchDropRollup(List<AlterClause> dropRollupClauses, Database
 
     public void processDropMaterializedView(DropMaterializedViewStmt dropMaterializedViewStmt, Database db,
             OlapTable olapTable) throws DdlException, MetaNotFoundException {
-        db.writeLock();
```

Comment:
Add a `Preconditions.checkState( db.isWriteLockHeldByCurrentThread())` to NOTICE.